tests/bluetooth/bsim_bt/compile.sh

```diff
@@ -39,8 +40,9 @@ function compile(){
       [ -d "${this_dir}" ] && rm ${this_dir} -rf
       mkdir -p ${this_dir} && cd ${this_dir}
       cmake -GNinja -DBOARD_ROOT=${BOARD_ROOT} -DBOARD=${BOARD} \
-            -DCONF_FILE=${conf_file} ${cmake_args} ${app_root}/${app} \
-            &> cmake.out || { cat cmake.out && return 0; }
+        -DCONF_FILE=${conf_file} ${cmake_args} \
+        -DCMAKE_C_FLAGS=${cc_flags} ${app_root}/${app} \
```

Comment:
You would need to quote it (`-DCMAKE_C_FLAGS="${cc_flags}"`) so multiple flags are supported

Reply:
Just compared 
```
  local cc_flags="${cc_flags:-"-Werror" "-Wall"}"

  echo "-DCMAKE_C_FLAGS=${cc_flags}"
```
```
alkh@alkh-virtual-linux:~/ncs/zephyr/tests/bluetooth/bsim_bt$ bash compile.sh 
-DCMAKE_C_FLAGS=-Werror -Wall
```
```
  local cc_flags="${cc_flags:-"-Werror" "-Wall"}"

  echo "-DCMAKE_C_FLAGS="${cc_flags}""
```
```
alkh@alkh-virtual-linux:~/ncs/zephyr/tests/bluetooth/bsim_bt$ bash compile.sh 
-DCMAKE_C_FLAGS=-Werror -Wall
```
No difference between optput

Reply:
That example is a bit missleading as echo just prints everything independently of spaces.
But when calling cmake, it is an argument, and an extra space without quoting results in 2 separate arguments. i.e. this is the exact call compile.sh does for the first app:
Without quotes:
```
cmake -GNinja -DBOARD_ROOT=zephyr -DBOARD=nrf52_bsim -DCONF_FILE=prj_split.conf -DCONFIG_COVERAGE=y -DCMAKE_C_FLAGS=-Wall -Werror zephyr/tests/bluetooth/bsim_bt/bsim_test_app
```
And with quotes:
```
cmake -GNinja -DBOARD_ROOT=zephyr -DBOARD=nrf52_bsim -DCONF_FILE=prj_split.conf -DCONFIG_COVERAGE=y '-DCMAKE_C_FLAGS=-Wall -Werror' zephyr/tests/bluetooth/bsim_bt/bsim_test_app
```
Note the single quote around the argument in the second case.
To try it, you can modify compile.sh (uncomment set -x, so you see what is executed in each case. You can also look at `zephyr/bsim_bt_out/tests/bluetooth/bsim_bt/bsim_test_app/prj_split.conf/build.ninja` to see the exact options that ended in the compiler call)